spanner/client.go

```diff
@@ -1439,3 +1471,31 @@ func logf(logger *log.Logger, format string, v ...interface{}) {
 		logger.Printf(format, v...)
 	}
 }
+
+// parseServerTimingHeader extracts server timing metrics from gRPC metadata into a map
+func parseServerTimingHeader(md metadata.MD) map[string]time.Duration {
+	metrics := make(map[string]time.Duration)
+	if md == nil {
+		return metrics
+	}
+
+	serverTiming := md.Get(serverTimingHeaderKey)
+	if len(serverTiming) == 0 {
+		return metrics
+	}
+
+	for _, timing := range serverTiming {
+		matches := serverTimingPattern.FindAllStringSubmatch(timing, -1)
+		for _, match := range matches {
+			if len(match) == 3 { // full match + 2 capture groups
+				metricName := match[1]
+				duration, err := strconv.ParseInt(match[2], 10, 64)
```

Comment:
Note: the duration could be float.

Reply:
Done, added unit test `TestParseServerTimingHeader`